packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterSerializedModel.kt

```diff
@@ -15,15 +15,18 @@
 
 package com.amazonaws.amplify.amplify_datastore.types.model
 
+import com.amplifyframework.core.model.Model
 import com.amplifyframework.core.model.temporal.Temporal
 import com.amplifyframework.datastore.appsync.SerializedModel
 
 
 data class FlutterSerializedModel(val serializedModel: SerializedModel) {
-    private val id: String = serializedModel.id
-    private val serializedData: Map<String, Any> = parseAnyMap(serializedModel.serializedData)
 
-    private val modelName: String = serializedModel.modelName
+    private val serializedData: Map<String, Any> = parseSerializedDataMap(serializedModel.serializedData)
+
+    // ignored fields
+    private val id: String = serializedModel.id
+    private val modelName: String = parseModelName(serializedModel.modelName) // ModelSchema -> SerializedModel should always have a name
```

Comment:
`serialiazedModel.modelName` can currently be null, in the case of a nested model (child).

Reply:
Indeed, the null value causes issues when serializing between Kotlin to Dart ...  Because of this I ensure that all values are empty strings rather than null.

Reply:
Somehow I'm not able to be convinced that this reason is strong enough to justify default empty strings. Will sync up offline over this.

Reply:
After speaking with @Amplifiyer we have decided to stop sending blank values in the Dart-> Kotlin map